macro/ShipAna.py

```
@@ -72,7 +72,11 @@
 
 run = ROOT.FairRunSim()
 run.SetName("TGeant4")  # Transport engine
-run.SetSink(ROOT.FairRootFileSink(ROOT.TMemFile("output", "recreate")))  # Dummy output file
+import tempfile
+
+sink = ROOT.FairRootFileSink(tempfile.mktemp(suffix=".root"))
```

Ruff (0.15.13)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)